src/main/java/org/jenkinsci/plugins/codesonar/api/CodeSonarHubAnalysisDataLoader.java

```diff
@@ -0,0 +1,238 @@
+        return visibilityFilter;
+    }
+
+    protected String getNewWarningsVisibilityFilter() {
+        return newWarningsVisibilityFilter;
+    }
+    
+    protected Analysis getLegacyAnalysisViewActive() throws IOException {
+        LOGGER.log(Level.INFO, "AnalysisViewActive not already set, loading from corresponding legacy service");
+        services.getAnalysisService().setVisibilityFilter(visibilityFilter);
+        return services.getAnalysisService().getAnalysisFromUrlWarningsByFilter(getBaseHubUri(), getAnalysisId());
+    }
+
+    protected Analysis getLegacyAnalysisViewNew() throws IOException {
+        LOGGER.log(Level.INFO, "AnalysisViewNew not already set, loading from corresponding legacy service");
+        services.getAnalysisService().setVisibilityFilter(newWarningsVisibilityFilter);
+        return services.getAnalysisService().getAnalysisFromUrlWithNewWarnings(getBaseHubUri(), getAnalysisId());
+    }
+
+    protected Procedures getLegacyProcedures() throws IOException {
+        LOGGER.log(Level.INFO, "Procedures not already set, loading from corresponding legacy service");
+        URI proceduresUri = services.getProceduresService().getProceduresUriFromAnAnalysisId(baseHubUri, String.valueOf(analysisId));
+        return services.getProceduresService().getProceduresFromUri(proceduresUri);
+    }
+    
+    protected Metrics getLegacyMetrics() throws IOException {
+        LOGGER.log(Level.INFO, "Metrics not already set, loading from corresponding legacy service");
+        URI metricsUri = services.getMetricsService().getMetricsUriFromAnAnalysisId(getBaseHubUri(), String.valueOf(analysisId));
+        return services.getMetricsService().getMetricsFromUri(metricsUri);
+    }
+    
+    public Long getNumberOfActiveWarnings() throws IOException {
+        LOGGER.log(Level.INFO, "getNumberOfActiveWarnings");
+        if(hubInfo.isJsonGridConfigSupported()) {
+            if(activeWarningsCount == null) {
+                LOGGER.log(Level.INFO, "ActiveWarningsCount not already set, loading from corresponding service");
+                activeWarningsCount = services.getAnalysisService().getNumberOfWarnings(getBaseHubUri(), getAnalysisId(), getVisibilityFilter());
+                LOGGER.log(Level.INFO, "ActiveWarningsCount new instance {0}", activeWarningsCount);
+                if(activeWarningsCount == null) {
+                    return null;
+                }
+            }
+            return activeWarningsCount.getNumberOfWarnings();
+        } else {
+            if(analysisViewActive == null) {
+                analysisViewActive = getLegacyAnalysisViewActive();
+                LOGGER.log(Level.INFO, "Legacy AnalysisViewActive new instance {0}", analysisViewActive);
+                if(analysisViewActive == null) {
+                    return null;
+                }
+            }
+            return (long) analysisViewActive.getWarnings().size();
+        }
+    }
```

Comment:
I would like to find a way to clean-up this `if (hubInfo.isJsonGridConfigSupported())` conditional pattern that appears in many of the methods in this data loader class.  Unfortunately, I think we are now spending too much time trying to refine the code, and I think we may have to live with this for the current version.  If we get into trouble with something else before we release, we may still want to address this, so I wouldn't dismiss the idea quite yet.  I think one way to solve the problem is with an inheritence mechanism.